### ISSUE TITLE
Navigation: Prioritize menu name and make it more prominent in the UI

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#59706]
### Description

Splitting this out from https://github.com/WordPress/gutenberg/pull/59440

On the front end, navigation menu must use a `<nav>` element. When there's more than one `<nav>` element (and this is likely the majority of the cases), each `<nav>` must be distinguished by the means of a meaningful `aria-label` attribute.
The label should not contain the word 'navigation' because screen readers alraedy announce a `<nav>` element as 'navigation'..

W3C tutorial and reference to success criteria:
https://www.w3.org/WAI/tutorials/page-structure/labels/

Instead, right now the editor ui:
- Deprioritizes the menu name by hiding it in most cases or placeing it in a dropdown menu.
- The UI doesn't explain to users that a menu name is extremly important because it is used on the front end for accessibility reasons adn as such it must be meaningful and unique.
- The UI doesn't allo users to name a menu while creating it. Instead, it implements a mechanism to automatically name menus that is less than ideal. For example, automatically created menu names could be:
  - 'Header navigation 3', used on the front end as: `aria-label="Header navigation 3"`
  - 'Footer navigation' used on the front end as `aria-label="Footer navigation"`.
- Renaming a menu is well buried down in the UI especially in the navigation block inspector, where it is placed under Block > Navigation > Gear icon (Settings) > Advanced (collapsed by default) > Menu name.

Given the fundamental importance of giving a menu a meaningful name, the current design that deprioritizes the menu name misses a functional requirement and it's far from ideal.

Instead of making the naming automatic and deprioritizing the menu name in the UI, the design shoudl be reversed.

- Naming a manu should be a first, prominent, step in the menu creation.
- The UI should explain to users why providing a meaningful name is important.
- The UI should explain to users why they have go avoid the word 'navigation' in the menu name.
- The menu name should be required.
- The menu name should be visually exposed in a prominent place wherever a menu editing UI is shown.
- The UI to rename a menu name should be very discoverable and placed in a prominent place.

Cc @joedolson @alexstine 

### Step-by-step reproduction instructions

N/A

### Screenshots, screen recording, code snippet

In the site editor navigation panel there is no way to creat a new menu. It's only meant to _manage_ existing menus.
Manu names are clearly exposed and the 'rename' action is in the 'Actions' allipsis dropwodn menu:

![Screenshot 2024-03-08 at 09 34 39](https://github.com/WordPress/gutenberg/assets/1682452/6be67ab5-45d5-477c-b132-4ac102f82780)

Instead, in the Navigation block inspector menu names are not visually exposed other than in the Actions dropdown menu. Newsly created menus are automatically named in a less than ideal way and the control to rename a menu is buried down in the UI:

![Screenshot 2024-03-08 at 11 03 51](https://github.com/WordPress/gutenberg/assets/1682452/115f6bdf-30fd-4264-a394-d325935fe9ff)

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by alexstine** [comment]
Funny enough, I actually worked on adding the numbered output to the `aria-label`. Did I appreciate that? No. Was it the solution I had at the time? Yes. Do I like this issue better? Absolutely. Does anything change from here? I guess that's up to the Design Team.

**Comment by jasmussen** [comment]
When you create a new page, you get this dialog:

![screenshot of the site editor  add new page  modal dialog, with title, cancel, and create draft button to confirm](https://github.com/WordPress/gutenberg/assets/1204802/3fa5add1-e8ec-4d96-9834-1ebafbe4311d)

Would a similar modal, shown regardless of where you invoke "Create new menu" help address the issue?

**Comment by afercia** [comment]
I think, an intemediary, required, step to provide a the menu name in a dialog would work.
The UI should explain _why_ a meaningful name is necessary though otherwise users may provide inappropriate / meaningless names.

**Comment by mrwweb** [comment]
Ran into this issue today and wanted to provide a bit more feedback.

As the OP notes, it's not clear how the Menu Name is used. What I think that also highlights is that there are likely cases where the Administrative Menu Name and the accessible, public-facing `aria-label` should be different!

In the case I was just working on, we're using the navigation block for an on-page table of contents. Given the context on the page, the label "Table of Contents" or "Page Sections" would be sufficient, but in the administrative site content, we need to be clearer, e.g. "Table of Contents for XXX".

However this is resolved, I think that have a "Menu Name" and "Accessible Navigation Name" or something similar is important.
